modules/connectors/ninja_van/tests/ninja_van/test_rate.py

```diff
@@ -0,0 +1,84 @@
+
+import unittest
+from unittest.mock import patch, ANY
+from .fixture import gateway
+
+import karrio
+import karrio.lib as lib
+import karrio.core.models as models
+
+
+class TestNinjaVanRating(unittest.TestCase):
+    def setUp(self):
+        self.maxDiff = None
+        self.RateRequest = models.RateRequest(**RatePayload)
+
+    def test_create_rate_request(self):
+        request = gateway.mapper.create_rate_request(self.RateRequest)
+
+        self.assertEqual(request.serialize(), RateRequest)
+
+    def test_get_rate(self):
+        with patch("karrio.mappers.ninja_van.proxy.lib.request") as mock:
+            mock.return_value = "{}"
+            karrio.Rating.fetch(self.RateRequest).from_(gateway)
+
+            self.assertEqual(
+                mock.call_args[1]["url"],
+                f"{gateway.settings.server_url}",
+            )
+
+    def test_parse_rate_response(self):
+        with patch("karrio.mappers.ninja_van.proxy.lib.request") as mock:
+            mock.return_value = RateResponse
+            parsed_response = karrio.Rating.fetch(self.RateRequest).from_(gateway).parse()
+
+            self.assertListEqual(lib.to_dict(parsed_response), ParsedRateResponse)
+
+
+if __name__ == "__main__":
+    unittest.main()
+
+
+RatePayload = {
+    "shipper": {
+        "company_name": "TESTING COMPANY",
+        "address_line1": "17 VULCAN RD",
+        "city": "CANNING VALE",
+        "postal_code": "6155",
+        "country_code": "AU",
+        "person_name": "TEST USER",
+        "state_code": "WA",
+        "email": "test@gmail.com",
+        "phone_number": "(07) 3114 1499",
+    },
+    "recipient": {
+        "company_name": "CGI",
+        "address_line1": "23 jardin private",
+        "city": "Ottawa",
+        "postal_code": "k1k 4t3",
+        "country_code": "CA",
+        "person_name": "Jain",
+        "state_code": "ON",
+    },
+    "parcels": [
+        {
+            "height": 50,
+            "length": 50,
+            "weight": 20,
+            "width": 12,
+            "dimension_unit": "CM",
+            "weight_unit": "KG",
+        }
+    ],
+    "options": { },
+    "reference": "REF-001",
+}
+
+ParsedRateResponse = []
+
+
+RateRequest = {}
+
+RateResponse = """{}
+"""
```

Comment:
For the tests, you want to run them and ensure that essentially when a request is created, you get the right output data that needs to be sent to NinjaVan 

Sendle is a good example here https://github.com/karrioapi/karrio/blob/main/modules/connectors/sendle/tests/sendle/test_rate.py